airbyte-cdk/python/unit_tests/sources/file_based/test_file_based_scenarios.py

```diff
@@ -152,7 +153,13 @@
 )
 from unit_tests.sources.file_based.test_scenarios import verify_check, verify_discover, verify_read, verify_spec
 
-discover_scenarios = [
+discover_failure_scenarios = [
+    earlier_csv_scenario,
+    empty_schema_inference_scenario,
+]
+
+discover_success_scenarios = [
+    csv_no_records_scenario,
```

Comment:
See other comment - not all discover scenarios can be used as read scenarios, as some that fail discover do not read successfully (in that they don't error successfully either)